web/js/containers/sidebar/smart-handoff.js

```diff
@@ -0,0 +1,470 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import * as olProj from 'ol/proj';
+import { debounce as lodashDebounce } from 'lodash';
+import moment from 'moment';
+import SmartHandoffModal from './smart-handoff-modal';
+import Button from '../../components/util/button';
+import Checkbox from '../../components/util/checkbox';
+import Crop from '../../components/util/image-crop';
+import util from '../../util/util';
+import { getLayers } from '../../modules/layers/selectors';
+import { imageUtilGetCoordsFromPixelValues } from '../../modules/image-download/util';
+import { onClose, openCustomContent } from '../../modules/modal/actions';
+import getSelectedDate from '../../modules/date/selectors';
+import safeLocalStorage from '../../util/local-storage';
+
+/**
+ * The Smart-Handoff components replaces the existing data download capability
+ * by directing users to select specific layer products within Worldview and 'hand' them 'off'
+ * to NASA's Earthdata Search web tool that will proceed in fetching corresponding
+ * layer data and granule files that are available for download.
+ */
+class SmartHandoff extends Component {
+  /**
+   * SmartHandoff's default constructor
+   * @param {*} props | A read-only object used to transfer data from a parent component
+   */
+  constructor(props) {
+    super(props);
+
+    const {
+      screenWidth,
+      screenHeight,
+    } = props;
+
+    // Set default state
+    this.state = {
+      boundaries: props.boundaries || {
+        x: screenWidth / 2 - 100,
+        y: screenHeight / 2 - 100,
+        x2: screenWidth / 2 + 100,
+        y2: screenHeight / 2 + 100,
+      },
+      selectedLayer: {},
+      showBoundingBox: false,
+      isSearchingForGranules: false,
+      selectedGranules: 0,
+      totalGranules: 0,
+      currentExtent: {},
+      coordinates: {},
+    };
+
+    this.baseState = this.state;
+    this.onBoundaryChange = this.onBoundaryChange.bind(this);
+    this.updateExtent = this.updateExtent.bind(this);
+    this.debouncedUpdateExtent = lodashDebounce(this.updateExtent, 250);
+  }
+
+  /**
+   * When fired, compare prevProps to determine if previously selected layer is still active
+   * and whether or not to update granule data base don data changes.
+   * @param {*} prevProps
+   */
+  componentDidUpdate(prevProps) {
+    const { selectedDate, activeLayers } = this.props;
+    const { currentExtent, selectedLayer } = this.state;
+
+    // Determine if existing selected layer is active still and visibility toggle is 'ON'
+    const isLayerStillActive = activeLayers.find((layer) => selectedLayer.conceptId === layer.conceptId);
```

Comment:
Totally optional but, a fun little trick to condense these one-liners is to deconstruct the single property you are using from the param, like so:
```suggestion
    const isLayerStillActive = activeLayers.find(({ conceptId }) => selectedLayer.conceptId === conceptId);
```

Reply:
Fancy.